src/sqlfluff/rules/L008.py

```diff
@@ -34,45 +36,76 @@ class Rule_L008(BaseRule):
         WHERE a IN ('plop',•'zoo')
     """
 
-    def _eval(self, context: RuleContext) -> Optional[LintResult]:
-        """Commas should be followed by a single whitespace unless followed by a comment.
+    def _get_subsequent_whitespace(
+        self,
+        context,
+    ) -> Tuple[Optional[BaseSegment], Optional[BaseSegment]]:
+        """Search forwards through the raw segments for subsequent whitespace.
 
-        This is a slightly odd one, because we'll almost always evaluate from a point a few places
-        after the problem site. NB: We need at least two segments behind us for this to work.
+        Return a tuple of both the trailing whitespace segment and the
+        first non-whitespace segment discovered.
         """
-        if len(context.raw_stack) < 1:
-            return None
+        subsequent_whitespace = None
+        # Get all raw segments and find position of the current comma within the list.
+        file_segment = context.parent_stack[0]
+        raw_segments = file_segment.get_raw_segments()
```

Comment:
Could you use `context.raw_stack` here instead? I think it may provide the same thing.

Reply:
@barrywhart `context.raw_stack` is different from `file_segment.get_raw_segments()`

`context.raw_stack` is a tuple of previously visited raw segments whereas `file_segment.get_raw_segments()` gives a list of all raw segments (including ones AFTER the current segment). This is important as we look ahead of the comma to see if there's whitespace so I think this is the correct approach

Reply:
Oh, of course... makes sense! 👍🏽